### PR TITLE
Initial documentation and README information

### PR DESCRIPTION
**Pull request by meew0**

### CODE REVIEW

File: src/discordcr/rest.cr [hunk 1]
```diff
@@ -13,6 +13,14 @@ module Discord
...
     alias RateLimitKey = {route_key: Symbol, major_parameter: UInt64?}
 
+    # Makes a REST request to Discord, with the given *method* to the given
+    # *path*, with the given *headers* set and with the given *body* being sent.
+    # The *route_key* should uniquely identify the route used, for rate limiting
+    # purposes. The *major_parameter* should be set to the guild or channel ID,
+    # if either of those appears as the first parameter in the route.
+    #
+    # This method also does rate limit handling, so if a rate limit is
+    # encountered, it may take longer than usual.
```
Comment (RX14) [inline src/discordcr/rest.cr]: Maybe mention that other events continue to be handled?

File: README.md [hunk 2]
```diff
@@ -2,8 +2,23 @@
 
 (The "cr" stands for "creative name".)
```
Comment (meew0) [inline README.md]: Why? :(

File: README.md [hunk 3]
```diff
@@ -13,29 +28,25 @@ dependencies:
...
-require "discordcr"
-```
-
-
-TODO: Write usage instructions here
-
-## Development
-
-TODO: Write development instructions here
+Documentation is available too but needs to be manually compiled locally.
```
Comment (meew0) [inline README.md]: Sure, go ahead.

File: src/discordcr/rest.cr (hunk 1 above)
Comment (meew0) [inline src/discordcr/rest.cr]: What do you mean? This is the REST part, no events are handled here.

Comment (RX14) [inline src/discordcr/rest.cr]: People who don't know crystal fibers well may think that this method blocking will stop events from being handled. It's only a very vague suggestion.

File: README.md [hunk 4]
```diff
@@ -49,4 +49,4 @@ will have the documentation to view.
 ## Contributors
 
 - [meew0](https://github.com/meew0) - creator, maintainer
-- [RX14](https://github.com/RX14) - co-developer, advisor
+- [RX14](https://github.com/RX14) - co-developer, advisor, Crystal expert, maintainer
```
Comment (RX14) [inline README.md]: I meant instead, not as well :)

co-developer sounds weird.

Comment (meew0) [inline README.md]: Fair enough.

File: README.md [hunk 5]
```diff
@@ -34,9 +34,42 @@ An example bot can be found
 [here](https://github.com/meew0/discordcr/blob/master/examples/ping.cr). More
 examples will come in the future.
 
-Documentation is available too but needs to be manually compiled locally.
-Download the repo and run `crystal doc` in its folder, then the `doc` folder
-will have the documentation to view.
+A short overview of library structure: the `Client` class includes the `REST`
+module, which handles the REST parts of Discord's API; itself it handles the
```
Comment (RX14) [inline README.md]: "...; the Client itself..."

File: README.md (hunk 2 above)
Comment (greenbigfrog) [inline README.md]: I vote for keep 😛

**Pull request merged**